packages/components/Tabs/src/TabsItem.tsx

```diff
@@ -39,24 +38,32 @@ export const TabsItem = compose<TabsItemType>({
     // Grabs the context information from Tabs (currently selected TabsItem and client's onTabsClick callback).
     const info = React.useContext(TabsContext);
 
-    /* There's a bug where the user callback is being called multiple times on one click.
-    We check that there is an actual change in selection before forwarding the message to the user callback
-    so that they aren't notified more than once for each click. */
-    const changeSelection = () => {
-      if (itemKey != info.selectedKey) {
+    const [focusState, setFocusState] = React.useState({
+      focused: false,
+    });
+
+    const changeSelection = React.useCallback(() => {
+      componentRef?.current?.focus();
+    }, [componentRef]);
+
+    const changeSelectionWithFocus = React.useCallback(() => {
+      setFocusState({ focused: true });
+      if (componentRef.current && !focusState.focused) {
```

Comment:
Wouldn't `focusState.focused = true` always? The line above, you set it to true.

Reply:
```setFocusState({ focused: true });``` works asynchronously. The first time that function is called, even if focusState.focused is set to true, it will be false on evaluation (from onBlur/removeFocus). The next evaluations done (repeated focus bug), focusState.focused wiill be true and stop the user functions to be called more than once.

Reply:
Is `focusState.focused = true` supposed to go on line 51?